integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -287,21 +273,19 @@ def export(self, **kwargs):  # pylint: disable=R0915
             enterprise_customer_identifier=self.enterprise_customer.name
         )
 
-        for enterprise_enrollment in enrollment_queryset:
-            is_audit_enrollment = enterprise_enrollment.is_audit_enrollment
-            if TransmissionAudit and skip_transmitted and \
-                    is_already_transmitted(TransmissionAudit, enterprise_enrollment.id, grade):
-                # We've already sent a completion status for this enrollment
-                generate_formatted_log(
-                    'Skipping export of previously sent enterprise enrollment. '
-                    'EnterpriseEnrollment: {enterprise_enrollment_id}'.format(
-                        enterprise_enrollment_id=enterprise_enrollment.id
-                    ),
-                    channel_name=channel_name,
-                    enterprise_customer_identifier=self.enterprise_customer.name
-                )
-                continue
+        if TransmissionAudit and skip_transmitted:
+            enrollments_to_transmit = self._filter_out_pre_transmitted_enrollments(
+                enrollment_queryset,
+                channel_name,
+                grade,
+                TransmissionAudit
+            )
+        else:
+            enrollments_to_transmit = enrollment_queryset
```

Comment:
The one thing that I don't like about this refactor is that your new _filter_out_pre_transmitted_enrollments function actually _resolves_ the query set (ie will make a database call) because it iterates over the results and this means that we could get slightly different behaviors at line 286, since if the code path goes through 284 it will have a different query.  That's one reason this was likely all kept together in the first place: your refactor will result in a hidden side effect that might make debugging later harder than needed.  

Basically, can you please change your _filter_out_pre_transmitted_enrollments to accept a _resolved_ queryset, so instead of passing it enrollment_queryset you pass it the result, like 'all_enrollments = enrollment_queryset.list()'  and then pass in 'all_enrollments'. And then you can change line 284 to also use 'all_enrollments' ?

Reply:
Further reading in this area: https://docs.djangoproject.com/en/3.1/ref/models/querysets/

Reply:
got it had to read it twice but yes I had a lingering thought in this area too, I think it's best to avoid the discrepancy.. will take a look thanks